src/plugins/vis_type_vega/public/data_model/search_api.ts

```diff
@@ -6,19 +6,38 @@
  * Side Public License, v 1.
  */
 
-import { combineLatest } from 'rxjs';
-import { map, tap } from 'rxjs/operators';
+import { combineLatest, from } from 'rxjs';
+import { map, tap, switchMap } from 'rxjs/operators';
 import { CoreStart, IUiSettingsClient } from 'kibana/public';
+import type { RuntimeFields } from '@elastic/elasticsearch/api/types';
+import { getData } from '../services';
 import {
   getSearchParamsFromRequest,
   SearchRequest,
   DataPublicPluginStart,
   IEsSearchResponse,
+  IndexPattern,
 } from '../../../data/public';
 import { search as dataPluginSearch } from '../../../data/public';
+import { ISearchRequestParams } from '../../../data/common';
 import { VegaInspectorAdapters } from '../vega_inspector';
 import { RequestResponder } from '../../../inspector/public';
 
+const extendSearchParamsWithRuntimeFields = (requestParams: ISearchRequestParams) =>
+  new Promise(async (resolve) => {
+    const indexPatternTitle = requestParams.index as string;
+    const indexPatterns = await getData().indexPatterns.find(indexPatternTitle);
+    const indexPattern = indexPatterns.find(
+      (index: IndexPattern) => index.title === indexPatternTitle
+    );
+    const runtimeFields = indexPattern?.getComputedFields().runtimeFields as RuntimeFields;
```

Comment:
are you sure you need `as RuntimeFields`? TypeScript helps you to make your code more stable. Let him do his job ;)